lib/Entities/StatisticsWooCommercePurchaseEntity.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace MailPoet\Entities;
+
+use MailPoet\Doctrine\EntityTraits\AutoincrementedIdTrait;
+use MailPoet\Doctrine\EntityTraits\CreatedAtTrait;
+use MailPoet\Doctrine\EntityTraits\UpdatedAtTrait;
+use MailPoetVendor\Doctrine\ORM\Mapping as ORM;
+
+/**
+ * @ORM\Entity()
+ * @ORM\Table(name="statistics_woocommerce_purchases")
+ */
+class StatisticsWooCommercePurchaseEntity {
+  use AutoincrementedIdTrait;
+  use CreatedAtTrait;
+  use UpdatedAtTrait;
+
+  /**
+   * @ORM\ManyToOne(targetEntity="MailPoet\Entities\NewsletterEntity")
+   * @ORM\JoinColumn(name="newsletter_id", referencedColumnName="id")
+   * @var NewsletterEntity|null
+   */
+  private $newsletter;
+
+  /**
+   * @ORM\ManyToOne(targetEntity="MailPoet\Entities\SendingQueueEntity")
+   * @ORM\JoinColumn(name="queue_id", referencedColumnName="id")
+   * @var SendingQueueEntity|null
+   */
+  private $queue;
+
+  /**
+   * @ORM\ManyToOne(targetEntity="MailPoet\Entities\SubscriberEntity")
+   * @ORM\JoinColumn(name="subscriber_id", referencedColumnName="id")
+   * @var SubscriberEntity|null
+   */
+  private $subscriber;
```

Comment:
I noticed that when we delete a subscriber we don't delete his purchases. I'm not sure what happens to this entity when the association refers to non-existing entity.

Reply:
It is probably ok for now since we don't work with the entity directly. It seems that when we delete subscriber (possible even other entities) we don't clean all related records and we leave some foreign keys. This may not pay off in the future. It makes sense to keep some anonymous statistic data after we delete a subscriber. Maybe we should make those relations nullable in DB and cleanup foreign keys when we delete a subscriber. So this is probably a more general problem not related directly to this PR.